core/src/main/resources/hudson/PluginManager/table.properties

```
@@ -32,16 +33,17 @@ javaWarning=\
  Warning: This plugin requires Java {0} or newer. \
  Jenkins will refuse to load this plugin if installed.
 depCompatWarning=\
- Warning: This plugin requires dependent plugins be upgraded and at least one of these dependent plugins claims to use a different settings format than the installed version. \
+ Warning: This plugin requires newer versions of dependencies and at least one of those plugins is not compatible with the installed version. \
+ This is usually the case because its behavior changed, or it uses a different settings format than the installed version. \
  Jobs using that plugin may need to be reconfigured, and/or you may not be able to cleanly revert to the prior version without manually restoring old settings. \
  Consult the plugin release notes for details.
 depCoreWarning=\
- Warning: This plugin requires dependent plugins that require Jenkins {0} or newer. \
- Jenkins will refuse to load the dependent plugins requiring a newer version of Jenkins, \
+ Warning: This plugin has dependencies to other plugins that require Jenkins {0} or newer. \
+ Jenkins will refuse to load the dependencies requiring a newer version of Jenkins, \
  and in turn loading this plugin will fail.
 depJavaWarning=\
- Warning: this plugin requires other plugins which in turn require Java {0} or newer. \
- Jenkins will refuse to load the dependent plugins requiring a newer version of Jenkins, \
+ Warning: this plugin has dependencies to other plugins that require Java {0} or newer. \
```

```suggestion
 Warning: this plugin has dependencies on other plugins that require Java {0} or newer. \
```

I am not a native speaker tho :)

---

I think this wording is better the other does work but doesn't read as well.